File: app/src/main/java/com/example/ikastyle/Common/Const/GearKind.java
package com.example.ikastyle.Common.Const;

/*
 * ギアの場所種別(アタマ,フク,クツ)のEnum
 */
public enum GearKind {
    HEAD(1), // アタマは1
    CLOTHING(2), // フクは2
    SHOES(3); // クツは3

    private int id;

    GearKind(int id){
        this.id = id;
    }

    public int getId(){
        return this.id;
    }

    // idを受け取って該当idのGearKindを返す
    public static GearKind getGearKind(int id){
        GearKind[] gearKinds = GearKind.values();
        for (GearKind item: gearKinds) {
            if(item.getId() == id){
                return item;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/splatool/ikastyle/model/common/const/GearKind.kt
package com.splatool.ikastyle.model.common.const

/*
 * ギアの場所種別(アタマ,フク,クツ)のEnum
 */
enum class GearKind(  // クツは3
    private val id: Int
) {
    HEAD(1),  // アタマは1
    CLOTHING(2),  // フクは2
    SHOES(3);

    fun getId(): Int {
        return id
    }

    companion object {
        // idを受け取って該当idのGearKindを返す
        fun getGearKind(id: Int): GearKind? {
            for(item: GearKind in values()){
                if (item.getId() == id) {
                    return item
                }
            }
            return null
        }
    }
}